src/qtt/measurements/new/signal_processor.py

```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+from qilib.data_set import DataSet
+
+from qtt.measurements.new.interfaces.signal_processor_interface import SignalProcessorInterface
+
+
+class SignalProcessor:
+    """ A class for post-processing measurement data. """
+
+    def __init__(self) -> None:
+        self._signal_processors: List[SignalProcessorInterface] = []
+
+    def add_signal_processor(self, signal_processor: SignalProcessorInterface) -> None:
+        """ Add a signal processor.
+
+            Args:
+                signal_processor: A signal processor
+        """
+
+        self._signal_processors.append(signal_processor)
+
+    def run_processes(self, signal_data: DataSet) -> DataSet:
```

Comment:
Again, naming could be better: `run_processes` suggest we fork things and create processes.. Which is not the case. Simply `run` would probably be enough, or if you want to be verbose `run_processors`, which is more accurate.

Reply:
Done